### PR TITLE
Wip/fix get harvester kubeconfig

### PR DESCRIPTION
**Pull request by m-ildefons**
#### Which issue(s) this PR fixes:

Issue # https://github.com/harvester/tests/issues/1627

#### What this PR does / why we need it:

Creating a cloud credential from a kubeconfig file that has not been created through Rancher is wrong, because then Rancher can't determine the expiration date of that kubeconfig and will return an error.
To fix this, the test suite needs to create the kubeconfig file through Rancher

### CODE REVIEW

**Review by FrankYang0529 — APPROVED** [review]

File: apiclient/rancher_api/api.py [hunk 1]
```diff
@@ -67,19 +67,27 @@ def __repr__(self):
 
     def _get(self, path, **kwargs):
         url = urljoin(self.endpoint, path)
-        return self.session.get(url, **kwargs)
+        resp = self.session.get(url, **kwargs)
+        print(f"{resp.request.method} {resp.request.url}")
```
Comment (lanfon72) [inline apiclient/rancher_api/api.py:71]: please remove the test output.

File: apiclient/rancher_api/managers.py [hunk 2]
```diff
@@ -678,6 +678,7 @@ def delete(self, name, *, raw=False):
 
 class ClusterManager(BaseManager):
     PATH_fmt = "v3/cluster/{uid}"
+    PATH1_fmt = "v3/clusters/{uid}"
```
Comment (lanfon72) [inline apiclient/rancher_api/managers.py:681]: it looks we didn't use this, is there any reason to add it?

File: harvester_e2e_tests/fixtures/settings.py [hunk 3]
```diff
@@ -15,7 +15,9 @@ def __init__(self):
 
         def _storage_net_configured(self):
             code, data = self.settings.get('storage-network')
-            if (cs := data.get('status', {}).get('conditions')):
```
Comment (lanfon72) [inline harvester_e2e_tests/fixtures/settings.py:18]: The assignment expressions (`:=`) is introduced in python 3.8, which is the version we targeting to support, is there any reason to replace it?

File: harvester_e2e_tests/integrations/test_9_rancher_integration.py [hunk 4]
```diff
@@ -130,7 +130,14 @@ def harvester_mgmt_cluster(api_client, rancher_api_client, unique_name, polling_
 @pytest.fixture(scope='module')
 def harvester_cloud_credential(api_client, rancher_api_client,
                                harvester_mgmt_cluster, unique_name):
-    harvester_kubeconfig = api_client.generate_kubeconfig()
+    code, data = rancher_api_client.clusters.generate_kubeconfig(
```
Comment (lanfon72) [inline harvester_e2e_tests/integrations/test_9_rancher_integration.py:133]: I think the two configurations are same, could you check with @albinsun about the change?

**Review by lanfon72 — COMMENTED** [review]
I think all changes are not harmful, but could you ask QAs to review PR rather than engineers next time?

and also cc @khushboo-rancher for your PR would be nice.

File: harvester_e2e_tests/fixtures/settings.py (hunk 3 above)
Comment (m-ildefons) [reply harvester_e2e_tests/fixtures/settings.py:18]: It's not supported in Python 3.6. We maintain a Python 3.6 environment in our tox config. Either we drop that as well, or we stick to Python 3.6 compatibility.

File: harvester_e2e_tests/integrations/test_9_rancher_integration.py (hunk 4 above)
Comment (m-ildefons) [reply harvester_e2e_tests/integrations/test_9_rancher_integration.py:133]: They are not the same. One uses Rancher's API endpoint and authenticates with a Rancher token and the other one uses Harvester's API endpoint directly.
For these tests we need to use Rancher's API endpoint.
Please read the issue linked up top.

**Comment by m-ildefons** [comment]
> I think all changes are not harmful, but could you ask QAs to review PR rather than engineers next time?

Your review was requested nearly a week ago.

**Pull request merged**